snap/validate.go

```diff
@@ -35,3 +35,13 @@ func ValidateName(name string) error {
 	}
 	return nil
 }
+
+// Validate verifies the content in the info.
+func Validate(info *Info) error {
```

Comment:
ValidateApp makes it easier to refactor minimally the code using verifyAppYaml atm and porting its tests, but yes we should decide wether:

1) always validate
2) validate only when we deal with (not yet installed snap) snap.yaml